lib/disposable.dart

```diff
@@ -37,13 +37,25 @@ class _InternalDisposable implements _Disposable {
   }
 }
 
+/// An interface to abstract management from actual disposal machinery.
+///
+/// When new management methods are to be added, they should be added
+/// here first, then implemented in [Disposable].
+abstract class DisposableManager {
+  void manageDisposable(Disposable disposable);
+  void manageDisposer(Disposer disposer);
+  void manageStreamController(StreamController controller);
+  void manageStreamSubscription(StreamSubscription subscription);
+}
+
 /// A function that, when called, disposes of one or more objects.
 typedef Future<dynamic> Disposer();
 
 /// Allows the creation of managed objects, including helpers for common patterns.
 ///
-/// There are three ways to consume this class: as a mixin, a base class,
-/// and an interface. All should work fine but the first is the simplest
+/// There are four ways to consume this class: as a mixin, a base class,
+/// an interface, and a concrete class used as a proxy. All should work
+/// fine but the first is the simplest
```

Comment:
That was already in there, but basically because it requires the least code on the part of the consumer.